Helm client: add methods that accept contexts

bkeroackdsc opened the pull request:
This adds methods to the helm client package (`pkg/helm`) that accept contexts. This allows callers to cancel long-running operations.

Fixes #4970

filipre committed:
Update control_structures.md (#4948)

update position of `and` operator

Signed-off-by: René Filip <rene.filip@sap.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

glb committed:
chore(deps): update cobra to latest (#4913)

Signed-off-by: Geoff Baskwill <me@geoffbaskwill.ca>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

bkeroackdsc committed:
helm client: add methods that accept contexts

Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

bkeroackdsc committed:
fix unit tests

Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

Commit:
fix(windows): fix unit tests on Windows (#4897)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

tariq1890 committed:
Adding link labels to doc links for more conspicuity (#4904)

Signed-off-by: tariqibrahim <tariq.ibrahim@microsoft.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

tao12345666333 committed:
Fix doc charts indent (#4940)

Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

a8uhnf committed:
Error message if anyone try to create multiple helm chart at a time (#4952)

* error message for creating multiple helm chart at a time

Signed-off-by: hanif <abuhanifa@pathao.com>

* Update cmd/helm/create.go

Co-Authored-By: a8uhnf <a8uhnf@users.noreply.github.com>
Signed-off-by: hanif <abuhanifa@pathao.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

OrthoDex committed:
docs: add documentation for the helmignore file (#4966)

* docs: add documentation for the helmignore file

Signed-off-by: Ishaan Malhi <ishaan.malhi@gmail.com>

* docs: rearrange helmignore docs under chart template developer's guide

Signed-off-by: Ishaan Malhi <ishaan.malhi@gmail.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

bacongobbler commented:
Hey, would you mind rebasing to ensure that the windows tests work? We fixed them up in #4897. Sorry about that, and thanks for the PR!

hickeyma committed:
Update chart doc with details about the license file (#4977)

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>

lukehoban committed:
Allow missing trailing '/' in --repo url (#4956)

Apply the same procedure to allow missing trailing slash in repo base URLs used in `repo/chart` inputs to `--repo` inputs.

Fixes #4954.

Signed-off-by: Luke Hoban <luke@pulumi.com>
Signed-off-by: Benjamen Keroack <benjamen.keroack@dollarshaveclub.com>
